Source/ProgressionSystemRuntime/Public/Data/PSDataAsset.h

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#pragma once
+
+#include "PSDataAsset.generated.h"
+
+/**
+ * Contains all progression assets used in the module 
+ */
+UCLASS()
+class PROGRESSIONSYSTEMRUNTIME_API UPSDataAsset : public UDataAsset
+{
+	GENERATED_BODY()
+
+public:
+	/** Returns the settings data asset. */
+	static const UPSDataAsset& Get() { return *GetDefault<ThisClass>(); }
```

Comment:
It will not work like this. `GetDefault` will return you CDO instance with default values of code class, but not values that changed in editor asset. Instead, you have to call `UPSWorldSystem::Get().GetPSDataAsset()` here. There is already comment with this task: [link](https://github.com/h4rdmol/ProgressionSystem/pull/1#discussion_r1493577275).

Reply:
This is resolved as part of improving the Get function with checkf. 
![image](https://github.com/h4rdmol/ProgressionSystem/assets/5227233/f53fb788-1c8b-41fb-a0b1-dd2c54d086a9)